### PR TITLE
feature: add support for AMQP

### PR DESCRIPTION
**Pull request by CodePrometheus**
Good evening friends!

[Amqp](https://github.com/rabbitmq/amqp091-go) is the Go Amqp Client Library officially maintained by the rabbitmq team.

I would be very grateful if a community member could review my code and provide some feedback.

### CODE REVIEW

**Comment by CodePrometheus** [comment]
Ref https://github.com/apache/skywalking/pull/11838

**Comment by wu-sheng** [comment]
@mrproliu is on vacation, this PR will be reviewed after he is back. Please wait for next week.

**Comment by wu-sheng** [comment]
@CodePrometheus Plugin docs should be updated. `support-plugins.md`.

**Comment by CodePrometheus** [comment]
Ok, I'll update the dos.
Since a new service was added to docker-composer this time, I found out through CI
`reason: [peer]: expected=>{amqp:5672}, actual=>{172.18.0.3:5672}`
My configuration is
```shell
amqp:
     image: rabbitmq:3.11.14-management
     hostname: amqp
     ...
```
Dear teachers, I would like to ask if I have any configuration errors

**Comment by CodePrometheus** [comment]
copy that, I'll try again locally

**Comment by CodePrometheus** [comment]
I used `SkyWalkingDynamicField` to deal with the peer value problem, and my locally env has passed the ./run.sh amqp test

File: CHANGES.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Release Notes.
 * Support setting a discard type of reporter.
 * Add `redis.max_args_bytes` parameter for redis plugin.
 * Changing intercept point for gin, make sure interfaces could be grouped when params defined in relativePath.
+* Support [Amqp](https://github.com/rabbitmq/amqp091-go) MQ.
```
Comment (wu-sheng) [inline CHANGES.md:14]:
```suggestion
* Support [AMQP](https://github.com/rabbitmq/amqp091-go) MQ.
```

File: docs/en/agent/support-plugins.md [hunk 2]
```diff
@@ -27,6 +27,8 @@ metrics based on the tracing data.
     * [MySQL Driver](https://github.com/go-sql-driver/mysql) tested v1.4.0 to v1.7.1.
 * Cache Client
   * `go-redisv9`: [go-redis](https://github.com/redis/go-redis) tested v9.0.3 to v9.0.5.
+* Cache Client
```
Comment (wu-sheng) [inline docs/en/agent/support-plugins.md:30]:
```suggestion
* MQ Client
```

File: test/plugins/scenarios/amqp/config/excepted.yml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            spanType: Entry
+            peer: ''
+            skipAnalysis: false
+            tags:
+              - {key: http.method, value: GET}
+              - {key: url, value: 'service:8080/execute'}
+              - {key: status_code, value: '200'}
+      - segmentId: not null
+        spans:
+          - operationName: Amqp/sw-queue/sw-consumer/Consumer
```
Comment (wu-sheng) [inline test/plugins/scenarios/amqp/config/excepted.yml:72]: Consumer should have reference from the producer side.

Comment (CodePrometheus) [reply test/plugins/scenarios/amqp/config/excepted.yml:72]: copy that, I'll fix it later

**Comment by mrproliu** [comment]
Please help to fix the comment.

**Comment by CodePrometheus** [comment]
terribly sorry, I've been stuck on a problem for a while, When I was dealing with consumer span, something didn't look right and there was no connection. I'm still trying to figure it out

image
image

**Comment by wu-sheng** [comment]
This is trace broken due to lack od reference of span for the entry(consumer) span.

Did you check context carrier injected into message header/metadata, and successfully extracted from it?

**Comment by CodePrometheus** [comment]
Thank you for the reminder, Teacher Wu. I checked that the context between the producer and consumer is connected, as shown in Figure 1. However, when I introduce an HTTP trigger, as shown in Figure 2, it appears that the amqp entrySpan `Amqp/...` is overriding the `GET:/consumer ` http entrySpan, I'm still trying to figure it out

image
image

**Comment by CodePrometheus** [comment]
> Are you using a HTTP handler to trigger consumer? I think you can't create another entry span when the first one not finished. You will need a local span in the middle, or use a timer to trigger consuming periodically.

Thanks, after several attempts, I think it should be able to meet our expectations.
Snipaste_2024-02-20_00-43-03

**Review by mrproliu — APPROVED** [review]
Thanks

**Comment by CodePrometheus** [comment]
Thanks all for your time as well.

**Pull request merged**